flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/source/IcebergSource.java

```diff
@@ -105,12 +107,12 @@ public class IcebergSource<T> implements Source<T, IcebergSourceSplit, IcebergEn
     this.readerFunction = readerFunction;
     this.assignerFactory = assignerFactory;
     this.splitComparator = splitComparator;
-    this.table = table;
     this.emitter = emitter;
+    this.tableName = table.name();
```

Comment:
this is incorrect. `table` can be null and loaded lazily.

Reply:
https://github.com/apache/iceberg/blob/8b7a280a9fd0b51eb43538fbb9f6879d1c92de18/flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/source/IcebergSource.java#L482

The builder method already handles the nullity case. I added a preconditions to verify that table is non-null before invoking the source constructor.

Reply:
Previously, users don't have to provide a valid table. this is a breaking change that requires users to supply a valid table object

Reply:
Thanks for pointing out the code from the `build()` method. yeah, this is fine. I missed it earlier